store/metric_test.go

```diff
@@ -0,0 +1,175 @@
+	suite.Suite
+	connURI      string
+	testDBName   string
+	mongoClient  *mongo.Client
+	testDatabase *mongo.Database
+}
+
+func NewMetricTestSuite(connURI, dbName string) *MetricTestSuite {
+	return &MetricTestSuite{
+		connURI:    connURI,
+		testDBName: dbName,
+	}
+}
+
+func (s *MetricTestSuite) SetupSuite() {
+	if s.connURI == "" || s.testDBName == "" {
+		s.T().Fatal("invalid test suite configuration")
+	}
+
+	opts := options.Client().ApplyURI(s.connURI)
+	mongoClient, err := mongo.NewClient(opts)
+	if nil != err {
+		s.T().Fatalf("create mongo client with error: %s", err)
+	}
+
+	if err = mongoClient.Connect(context.Background()); nil != err {
+		s.T().Fatalf("connect mongo database with error: %s", err.Error())
+	}
+
+	s.mongoClient = mongoClient
+	s.testDatabase = mongoClient.Database(s.testDBName)
+
+	// make sure the test suite is run with a clean environment
+	if err := s.CleanMongoDB(); err != nil {
+		s.T().Fatal(err)
+	}
+	schema.NewMongoDBIndexer(s.connURI, s.testDBName).IndexAll()
+	if err := s.LoadMongoDBFixtures(); err != nil {
+		s.T().Fatal(err)
+	}
+}
+
+// LoadMongoDBFixtures will preload fixtures into test mongodb
+func (s *MetricTestSuite) LoadMongoDBFixtures() error {
+	ctx := context.Background()
+
+	if _, err := s.testDatabase.Collection(schema.ProfileCollection).InsertMany(ctx, []interface{}{
+		schema.Profile{
+			ID:            "test-account-profile-id",
+			AccountNumber: "account-test",
+		},
+	}); err != nil {
+		return err
+	}
+
+	if _, err := s.testDatabase.Collection(schema.SymptomReportCollection).InsertMany(ctx, []interface{}{
+		metricTestSymptomReport1,
+		metricTestSymptomReport2,
+		metricTestSymptomReport3,
+	}); err != nil {
+		s.T().Fatal(err)
+	}
+
+	if _, err := s.testDatabase.Collection(schema.BehaviorReportCollection).InsertMany(ctx, []interface{}{
+		metricTestBehaviorReport1,
+		metricTestBehaviorReport2,
+		metricTestBehaviorReport3,
+	}); err != nil {
+		s.T().Fatal(err)
+	}
+
+	return nil
+}
+
+// CleanMongoDB drop the whole test mongodb
+func (s *MetricTestSuite) CleanMongoDB() error {
+	return s.testDatabase.Drop(context.Background())
+}
+
+// TestSyncProfileIndividualMetrics tests if the function returned values
+// are identical to the values in database.
+func (s *MetricTestSuite) TestSyncProfileIndividualMetrics() {
+	store := NewMongoStore(s.mongoClient, s.testDBName)
+
+	m, err := store.SyncProfileIndividualMetrics("test-account-profile-id")
+	s.NoError(err)
+
+	var profile schema.Profile
+	err = s.testDatabase.Collection(schema.ProfileCollection).FindOne(context.Background(), bson.M{
+		"id": "test-account-profile-id",
+	}, options.FindOne().SetProjection(bson.M{"individual_metric": 1})).Decode(&profile)
+	s.NoError(err)
+
+	s.Equal(profile.IndividualMetric.SymptomCount, m.SymptomCount)
+	s.Equal(profile.IndividualMetric.SymptomDelta, m.SymptomDelta)
+	s.Equal(profile.IndividualMetric.BehaviorCount, m.BehaviorCount)
+	s.Equal(profile.IndividualMetric.BehaviorDelta, m.BehaviorDelta)
```

Comment:
Shall we also verify if the values are correct?
`s.Equal(profile.IndividualMetric.BehaviorCount, 6)`

Reply:
I originally did. However, it mostly calls calculating functions which has already had test cases.